src/schema/posts.ts

```diff
@@ -1529,11 +1582,8 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
           ),
         (builder) => {
           builder.queryBuilder
-            .where(`"${builder.alias}"."sourceId" = :sourceId`, {
-              sourceId: args.sourceId,
-            })
-            .andWhere(`"${builder.alias}"."createdById" = :userId`, {
-              userId,
+            .where(`"${builder.alias}"."sourceId" IN (:...sourceIds)`, {
```

Comment:
I think you should be able to do this in a single query with join over source member and then filter by role? No need to first query `sourceIds` above.